php/Factory.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace PHPCD;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Lvht\MsgpackRpc\JsonMessenger;
+use Lvht\MsgpackRpc\Io;
+use Lvht\MsgpackRpc\MsgpackMessenger;
+
+/**
+ * Simple factory to separate details of object creation
+ */
+class Factory
+{
+    /**
+     * @return ContainerBuilder
+     */
+    public function createDIContainer($configFileName, $configDir, $additionalParameters = [])
+    {
+        $container = new ContainerBuilder;
+
+        $loader = new YamlFileLoader($container, new FileLocator($configDir));
```

Comment:
It is used only at the stage of resolving dependencies and not used when both daemons already works so IMHO it does not affect performance of normal work. On the other side, YAML notation is more readable and easier to maintain than its bare PHP equivalent (`$container->setDefinition(....`).